db/db_impl/db_impl.cc

```diff
@@ -422,16 +414,31 @@ Status DBImpl::ResumeImpl(DBRecoverContext context) {
     // during previous error handling.
     if (file_deletion_disabled) {
       // Always return ok
-      s = EnableFileDeletions(/*force=*/true);
-      if (!s.ok()) {
+      Status enable_file_deletion_s = EnableFileDeletions(/*force=*/true);
+      if (!enable_file_deletion_s.ok()) {
         ROCKS_LOG_INFO(
             immutable_db_options_.info_log,
             "DB resume requested but could not enable file deletions [%s]",
-            s.ToString().c_str());
+            enable_file_deletion_s.ToString().c_str());
       }
     }
+  }
+
+  mutex_.Lock();
+  if (s.ok()) {
+    s = error_handler_.ClearBGError();
```

Comment:
Maybe add some comment here about how the db may be closed?

Reply:
Fixed